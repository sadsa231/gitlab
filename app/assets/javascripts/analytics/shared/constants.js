import { masks } from '~/lib/dateformat';
import { s__ } from '~/locale';
import { helpPagePath } from '~/helpers/help_page_helper';

export const DATE_RANGE_LIMIT = 180;
export const PROJECTS_PER_PAGE = 50;

const { isoDate, mediumDate } = masks;
export const dateFormats = {
  isoDate,
  defaultDate: mediumDate,
  defaultDateTime: 'mmm d, yyyy h:MMtt',
  month: 'mmmm',
};

export const KEY_METRICS = {
  LEAD_TIME: 'lead_time',
  CYCLE_TIME: 'cycle_time',
  ISSUES: 'issues',
  COMMITS: 'commits',
  DEPLOYS: 'deploys',
};

export const DORA_METRICS = {
  DEPLOYMENT_FREQUENCY: 'deployment_frequency',
  LEAD_TIME_FOR_CHANGES: 'lead_time_for_changes',
  TIME_TO_RESTORE_SERVICE: 'time_to_restore_service',
  CHANGE_FAILURE_RATE: 'change_failure_rate',
};

export const VSA_METRICS_GROUPS = [
  {
    key: 'key_metrics',
    title: s__('ValueStreamAnalytics|Key metrics'),
    keys: Object.values(KEY_METRICS),
  },
  {
    key: 'dora_metrics',
    title: s__('ValueStreamAnalytics|DORA metrics'),
    keys: Object.values(DORA_METRICS),
  },
];

export const METRIC_TOOLTIPS = {
  [DORA_METRICS.DEPLOYMENT_FREQUENCY]: {
    description: s__(
      'ValueStreamAnalytics|Average number of deployments to production per day. This metric measures how often value is delivered to end users.',
    ),
    groupLink: '-/analytics/ci_cd?tab=deployment-frequency',
    projectLink: '-/pipelines/charts?chart=deployment-frequency',
    docsLink: helpPagePath('user/analytics/dora_metrics', { anchor: 'deployment-frequency' }),
  },
  [DORA_METRICS.LEAD_TIME_FOR_CHANGES]: {
    description: s__(
      'ValueStreamAnalytics|The time to successfully deliver a commit into production. This metric reflects the efficiency of CI/CD pipelines.',
    ),
    groupLink: '-/analytics/ci_cd?tab=lead-time',
    projectLink: '-/pipelines/charts?chart=lead-time',
    docsLink: helpPagePath('user/analytics/dora_metrics', { anchor: 'lead-time-for-changes' }),
  },
  [DORA_METRICS.TIME_TO_RESTORE_SERVICE]: {
    description: s__(
      'ValueStreamAnalytics|The time it takes an organization to recover from a failure in production.',
    ),
    groupLink: '-/analytics/ci_cd?tab=time-to-restore-service',
    projectLink: '-/pipelines/charts?chart=time-to-restore-service',
    docsLink: helpPagePath('user/analytics/dora_metrics', { anchor: 'time-to-restore-service' }),
  },
  [DORA_METRICS.CHANGE_FAILURE_RATE]: {
    description: s__(
      'ValueStreamAnalytics|Percentage of deployments that cause an incident in production.',
    ),
    groupLink: '-/analytics/ci_cd?tab=change-failure-rate',
    projectLink: '-/pipelines/charts?chart=change-failure-rate',
    docsLink: helpPagePath('user/analytics/dora_metrics', { anchor: 'change-failure-rate' }),
  },
  [KEY_METRICS.LEAD_TIME]: {
    description: s__('ValueStreamAnalytics|Median time from issue created to issue closed.'),
    groupLink: '-/analytics/value_stream_analytics',
    projectLink: '-/value_stream_analytics',
    docsLink: helpPagePath('user/analytics/value_stream_analytics', {
      anchor: 'view-the-lead-time-and-cycle-time-for-issues',
    }),
  },
  [KEY_METRICS.CYCLE_TIME]: {
    description: s__(
      "ValueStreamAnalytics|Median time from the earliest commit of a linked issue's merge request to when that issue is closed.",
    ),
    groupLink: '-/analytics/value_stream_analytics',
    projectLink: '-/value_stream_analytics',
    docsLink: helpPagePath('user/analytics/value_stream_analytics', {
      anchor: 'view-the-lead-time-and-cycle-time-for-issues',
    }),
  },
  [KEY_METRICS.ISSUES]: {
    description: s__('ValueStreamAnalytics|Number of new issues created.'),
    groupLink: '-/issues_analytics',
    projectLink: '-/analytics/issues_analytics',
    docsLink: helpPagePath('user/analytics/issue_analytics'),
  },
  [KEY_METRICS.DEPLOYS]: {
    description: s__('ValueStreamAnalytics|Total number of deploys to production.'),
    groupLink: '-/analytics/productivity_analytics',
    projectLink: '-/analytics/merge_request_analytics',
    docsLink: helpPagePath('user/analytics/merge_request_analytics'),
  },
};

// TODO: Remove this once the migration to METRIC_TOOLTIPS is complete
// https://gitlab.com/gitlab-org/gitlab/-/issues/388067
export const METRICS_POPOVER_CONTENT = {
  lead_time: {
    description: s__('ValueStreamAnalytics|Median time from issue created to issue closed.'),
  },
  cycle_time: {
    description: s__(
      "ValueStreamAnalytics|Median time from the earliest commit of a linked issue's merge request to when that issue is closed.",
    ),
  },
  lead_time_for_changes: {
    description: s__(
      'ValueStreamAnalytics|Median time between merge request merge and deployment to a production environment for all MRs deployed in the given time period.',
    ),
  },
  issues: { description: s__('ValueStreamAnalytics|Number of new issues created.') },
  deploys: { description: s__('ValueStreamAnalytics|Total number of deploys to production.') },
  deployment_frequency: {
    description: s__('ValueStreamAnalytics|Average number of deployments to production per day.'),
  },
  commits: {
    description: s__('ValueStreamAnalytics|Number of commits pushed to the default branch'),
  },
  time_to_restore_service: {
    description: s__(
      'ValueStreamAnalytics|Median time an incident was open on a production environment in the given time period.',
    ),
  },
  change_failure_rate: {
    description: s__(
      'ValueStreamAnalytics|Percentage of deployments that cause an incident in production.',
    ),
  },
};
