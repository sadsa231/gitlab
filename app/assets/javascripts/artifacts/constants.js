import { __, s__, n__, sprintf } from '~/locale';

export const JOB_STATUS_GROUP_SUCCESS = 'success';

export const STATUS_BADGE_VARIANTS = {
  success: 'success',
  passed: 'success',
  error: 'danger',
  failed: 'danger',
  pending: 'warning',
  'waiting-for-resource': 'warning',
  'failed-with-warnings': 'warning',
  'success-with-warnings': 'warning',
  running: 'info',
  canceled: 'neutral',
  disabled: 'neutral',
  scheduled: 'neutral',
  manual: 'neutral',
  notification: 'muted',
  preparing: 'muted',
  created: 'muted',
  skipped: 'muted',
  notfound: 'muted',
};

export const I18N_DOWNLOAD = __('Download');
export const I18N_BROWSE = s__('Artifacts|Browse');
export const I18N_DELETE = __('Delete');
export const I18N_EXPIRED = __('Expired');
export const I18N_DESTROY_ERROR = s__('Artifacts|An error occurred while deleting the artifact');
export const I18N_FETCH_ERROR = s__('Artifacts|An error occurred while retrieving job artifacts');
export const I18N_ARTIFACTS = __('Artifacts');
export const I18N_JOB = __('Job');
export const I18N_SIZE = __('Size');
export const I18N_CREATED = __('Created');
export const I18N_ARTIFACTS_COUNT = (count) => n__('%d file', '%d files', count);

export const I18N_MODAL_TITLE = (artifactName) =>
  sprintf(s__('Artifacts|Delete %{name}?'), { name: artifactName });
export const I18N_MODAL_BODY = s__(
  'Artifacts|This artifact will be permanently deleted. Any reports generated from this artifact will be empty.',
);
export const I18N_MODAL_PRIMARY = s__('Artifacts|Delete artifact');
export const I18N_MODAL_CANCEL = __('Cancel');

export const I18N_FEEDBACK_BANNER_TITLE = s__('Artifacts|Help us improve this page');
export const I18N_FEEDBACK_BANNER_BODY = s__(
  'Artifacts|We want you to be able to use this page to easily manage your CI/CD job artifacts. We are working to improve this experience and would appreciate any feedback you have about the improvements we are making.',
);
export const I18N_FEEDBACK_BANNER_BUTTON = s__('Artifacts|Take a quick survey');
export const FEEDBACK_URL = 'https://gitlab.fra1.qualtrics.com/jfe/form/SV_cI9rAUI20Vo2St8';

export const INITIAL_CURRENT_PAGE = 1;
export const INITIAL_PREVIOUS_PAGE_CURSOR = '';
export const INITIAL_NEXT_PAGE_CURSOR = '';
export const JOBS_PER_PAGE = 20;
export const INITIAL_LAST_PAGE_SIZE = null;

export const ARCHIVE_FILE_TYPE = 'ARCHIVE';

export const ARTIFACT_ROW_HEIGHT = 56;
export const ARTIFACTS_SHOWN_WITHOUT_SCROLLING = 4;
