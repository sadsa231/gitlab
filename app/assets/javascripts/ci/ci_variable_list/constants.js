import { __, s__ } from '~/locale';

export const ADD_CI_VARIABLE_MODAL_ID = 'add-ci-variable';

// This const will be deprecated once we remove VueX from the section
export const displayText = {
  variableText: __('Variable'),
  fileText: __('File'),
  allEnvironmentsText: __('All (default)'),
};

export const variableTypes = {
  envType: 'ENV_VAR',
  fileType: 'FILE',
};

// Once REST is removed, we won't need `types`
export const types = {
  variableType: 'env_var',
  fileType: 'file',
};

export const allEnvironments = {
  type: '*',
  text: __('All (default)'),
};

// Once REST is removed, we won't need `types` key
export const variableText = {
  [types.variableType]: __('Variable'),
  [types.fileType]: __('File'),
  [variableTypes.envType]: __('Variable'),
  [variableTypes.fileType]: __('File'),
};

export const variableOptions = [
  { value: variableTypes.envType, text: variableText[variableTypes.envType] },
  { value: variableTypes.fileType, text: variableText[variableTypes.fileType] },
];

export const defaultVariableState = {
  environmentScope: allEnvironments.type,
  key: '',
  masked: false,
  protected: false,
  raw: false,
  value: '',
  variableType: variableTypes.envType,
};

// eslint-disable-next-line @gitlab/require-i18n-strings
export const groupString = 'Group';
// eslint-disable-next-line @gitlab/require-i18n-strings
export const instanceString = 'Instance';
// eslint-disable-next-line @gitlab/require-i18n-strings
export const projectString = 'Project';

export const AWS_TIP_DISMISSED_COOKIE_NAME = 'ci_variable_list_constants_aws_tip_dismissed';
export const AWS_TIP_MESSAGE = __(
  '%{deployLinkStart}Use a template to deploy to ECS%{deployLinkEnd}, or use a docker image to %{commandsLinkStart}run AWS commands in GitLab CI/CD%{commandsLinkEnd}.',
);

export const EVENT_LABEL = 'ci_variable_modal';
export const EVENT_ACTION = 'validation_error';

// AWS TOKEN CONSTANTS
export const AWS_ACCESS_KEY_ID = 'AWS_ACCESS_KEY_ID';
export const AWS_DEFAULT_REGION = 'AWS_DEFAULT_REGION';
export const AWS_SECRET_ACCESS_KEY = 'AWS_SECRET_ACCESS_KEY';
export const AWS_TOKEN_CONSTANTS = [AWS_ACCESS_KEY_ID, AWS_DEFAULT_REGION, AWS_SECRET_ACCESS_KEY];

export const CONTAINS_VARIABLE_REFERENCE_MESSAGE = __(
  'Unselect "Expand variable reference" if you want to use the variable value as a raw string.',
);
export const DEFAULT_EXCEEDS_VARIABLE_LIMIT_TEXT = s__(
  'CiVariables|You have reached the maximum number of variables available. To add new variables, you must reduce the number of defined variables.',
);
export const ENVIRONMENT_SCOPE_LINK_TITLE = __('Learn more');
export const EXCEEDS_VARIABLE_LIMIT_TEXT = s__(
  'CiVariables|This %{entity} has %{currentVariableCount} defined CI/CD variables. The maximum number of variables per %{entity} is %{maxVariableLimit}. To add new variables, you must reduce the number of defined variables.',
);
export const FLAG_LINK_TITLE = s__('CiVariable|Define a CI/CD variable in the UI');
export const MAXIMUM_VARIABLE_LIMIT_REACHED = s__(
  'CiVariables|Maximum number of variables reached.',
);

export const ADD_VARIABLE_ACTION = 'ADD_VARIABLE';
export const EDIT_VARIABLE_ACTION = 'EDIT_VARIABLE';
export const VARIABLE_ACTIONS = [ADD_VARIABLE_ACTION, EDIT_VARIABLE_ACTION];

export const GRAPHQL_PROJECT_TYPE = 'Project';
export const GRAPHQL_GROUP_TYPE = 'Group';

export const ADD_MUTATION_ACTION = 'add';
export const UPDATE_MUTATION_ACTION = 'update';
export const DELETE_MUTATION_ACTION = 'delete';

export const EXPANDED_VARIABLES_NOTE = __(
  '%{codeStart}$%{codeEnd} will be treated as the start of a reference to another variable.',
);

export const environmentFetchErrorText = __(
  'There was an error fetching the environments information.',
);
export const genericMutationErrorText = __('Something went wrong on our end. Please try again.');
export const variableFetchErrorText = __('There was an error fetching the variables.');
