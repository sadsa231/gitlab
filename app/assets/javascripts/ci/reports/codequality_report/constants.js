export const SEVERITY_CLASSES = {
  info: 'text-primary-400',
  minor: 'text-warning-200',
  major: 'text-warning-400',
  critical: 'text-danger-600',
  blocker: 'text-danger-800',
  unknown: 'text-secondary-400',
};

export const SEVERITY_ICONS = {
  info: 'severity-info',
  minor: 'severity-low',
  major: 'severity-medium',
  critical: 'severity-high',
  blocker: 'severity-critical',
  unknown: 'severity-unknown',
};

export const SEVERITY_ICONS_MR_WIDGET = {
  info: 'severityInfo',
  minor: 'severityLow',
  major: 'severityMedium',
  critical: 'severityHigh',
  blocker: 'severityCritical',
  unknown: 'severityUnknown',
};

export const SEVERITIES = {
  info: {
    class: SEVERITY_CLASSES.info,
    name: SEVERITY_ICONS.info,
  },
  minor: {
    class: SEVERITY_CLASSES.minor,
    name: SEVERITY_ICONS.minor,
  },
  major: {
    class: SEVERITY_CLASSES.major,
    name: SEVERITY_ICONS.major,
  },
  critical: {
    class: SEVERITY_CLASSES.critical,
    name: SEVERITY_ICONS.critical,
  },
  blocker: {
    class: SEVERITY_CLASSES.blocker,
    name: SEVERITY_ICONS.blocker,
  },
  unknown: {
    class: SEVERITY_CLASSES.unknown,
    name: SEVERITY_ICONS.unknown,
  },
};
