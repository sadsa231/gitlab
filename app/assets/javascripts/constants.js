/* eslint-disable @gitlab/require-i18n-strings */

export const getModifierKey = (removeSuffix = false) => {
  const winKey = `Ctrl${removeSuffix ? '' : '+'}`;
  return window.gl?.client?.isMac ? 'âŒ˜' : winKey;
};
