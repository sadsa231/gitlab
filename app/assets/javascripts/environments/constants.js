import { __, s__ } from '~/locale';
import { getDateInPast } from '~/lib/utils/datetime_utility';

// These statuses are based on how the backend defines pod phases here
// lib/gitlab/kubernetes/pod.rb

export const STATUS_MAP = {
  succeeded: {
    class: 'succeeded',
    text: __('Succeeded'),
    stable: true,
  },
  running: {
    class: 'running',
    text: __('Running'),
    stable: true,
  },
  failed: {
    class: 'failed',
    text: __('Failed'),
    stable: true,
  },
  pending: {
    class: 'pending',
    text: __('Pending'),
    stable: true,
  },
  unknown: {
    class: 'unknown',
    text: __('Unknown'),
    stable: true,
  },
};

export const CANARY_STATUS = {
  class: 'canary-icon',
  text: __('Canary'),
  stable: false,
};

export const CANARY_UPDATE_MODAL = 'confirm-canary-change';

export const ENVIRONMENTS_SCOPE = {
  AVAILABLE: 'available',
  STOPPED: 'stopped',
};

export const ENVIRONMENT_COUNT_BY_SCOPE = {
  [ENVIRONMENTS_SCOPE.AVAILABLE]: 'availableCount',
  [ENVIRONMENTS_SCOPE.STOPPED]: 'stoppedCount',
};

export const REVIEW_APP_MODAL_I18N = {
  title: s__('ReviewApp|Enable Review App'),
  intro: s__(
    'EnableReviewApp|Review apps are dynamic environments that you can use to provide a live preview of changes made in a feature branch.',
  ),
  instructions: {
    title: s__('EnableReviewApp|To configure a dynamic review app, you must:'),
    step1: s__(
      'EnableReviewApp|Have access to infrastructure that can host and deploy the review apps.',
    ),
    step2: s__('EnableReviewApp|Install and configure a runner to do the deployment.'),
    step3: s__('EnableReviewApp|Add a job in your CI/CD configuration that:'),
    step3a: s__('EnableReviewApp|Only runs for feature branches or merge requests.'),
    step3b: s__(
      'EnableReviewApp|Uses a predefined CI/CD variable like %{codeStart}$(CI_COMMIT_REF_SLUG)%{codeEnd} to dynamically create the review app environments. For example, for a configuration using merge request pipelines:',
    ),
    step4: s__('EnableReviewApp|Recommended: Set up a job that manually stops the Review Apps.'),
  },
  staticSitePopover: {
    title: s__('EnableReviewApp|Using a static site?'),
    body: s__(
      'EnableReviewApp|Make sure your project has an environment configured with the target URL set to your website URL. If not, create a new one before continuing.',
    ),
  },
  learnMore: __('Learn more'),
  viewMoreExampleProjects: s__('EnableReviewApp|View more example projects'),
  copyToClipboardText: s__('EnableReviewApp|Copy snippet'),
};

export const MIN_STALE_ENVIRONMENT_DATE = getDateInPast(new Date(), 3650); // 10 years ago
export const MAX_STALE_ENVIRONMENT_DATE = getDateInPast(new Date(), 7); // one week ago

export const ENVIRONMENT_NEW_HELP_TEXT = __(
  'Environments allow you to track deployments of your application.%{linkStart} More information.%{linkEnd}',
);

export const ENVIRONMENT_EDIT_HELP_TEXT = ENVIRONMENT_NEW_HELP_TEXT;
