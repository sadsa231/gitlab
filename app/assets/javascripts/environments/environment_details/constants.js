import { __, s__ } from '~/locale';
import { helpPagePath } from '~/helpers/help_page_helper';

export const ENVIRONMENT_DETAILS_PAGE_SIZE = 20;
export const ENVIRONMENT_DETAILS_TABLE_FIELDS = [
  {
    key: 'status',
    label: __('Status'),
    columnClass: 'gl-w-10p',
    tdClass: 'gl-vertical-align-middle!',
  },
  {
    key: 'id',
    label: __('ID'),
    columnClass: 'gl-w-5p',
    tdClass: 'gl-vertical-align-middle!',
  },
  {
    key: 'triggerer',
    label: __('Triggerer'),
    columnClass: 'gl-w-10p',
    tdClass: 'gl-vertical-align-middle!',
  },
  {
    key: 'commit',
    label: __('Commit'),
    columnClass: 'gl-w-20p',
    tdClass: 'gl-vertical-align-middle!',
  },
  {
    key: 'job',
    label: __('Job'),
    columnClass: 'gl-w-20p',
    tdClass: 'gl-vertical-align-middle!',
  },
  {
    key: 'created',
    label: __('Created'),
    columnClass: 'gl-w-10p',
    tdClass: 'gl-vertical-align-middle! gl-white-space-nowrap',
  },
  {
    key: 'deployed',
    label: __('Deployed'),
    columnClass: 'gl-w-10p',
    tdClass: 'gl-vertical-align-middle! gl-white-space-nowrap',
  },
  {
    key: 'actions',
    label: __('Actions'),
    columnClass: 'gl-w-10p',
    tdClass: 'gl-vertical-align-middle! gl-white-space-nowrap',
  },
];

export const translations = {
  emptyStateTitle: s__("Deployments|You don't have any deployments right now."),
  emptyStatePrimaryButton: __('Read more'),
  emptyStateDescription: s__(
    'Deployments|Define environments in the deploy stage(s) in %{code_open}.gitlab-ci.yml%{code_close} to track deployments here.',
  ),
  nextPageButtonLabel: __('Next'),
  previousPageButtonLabel: __('Prev'),
};

export const codeBlockPlaceholders = { code: ['code_open', 'code_close'] };

export const environmentsHelpPagePath = helpPagePath('ci/environments/index.md');
