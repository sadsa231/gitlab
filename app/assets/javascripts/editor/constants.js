import { getModifierKey } from '~/constants';
import { DEFAULT_DEBOUNCE_AND_THROTTLE_MS } from '~/lib/utils/constants';
import { s__, __, sprintf } from '~/locale';

const modifierKey = getModifierKey();

export const URI_PREFIX = 'gitlab';
export const CONTENT_UPDATE_DEBOUNCE = DEFAULT_DEBOUNCE_AND_THROTTLE_MS;

export const EDITOR_READY_EVENT = 'editor-ready';

export const EDITOR_TYPE_CODE = 'vs.editor.ICodeEditor';
export const EDITOR_TYPE_DIFF = 'vs.editor.IDiffEditor';

export const EDITOR_CODE_INSTANCE_FN = 'createInstance';
export const EDITOR_DIFF_INSTANCE_FN = 'createDiffInstance';

export const EDITOR_TOOLBAR_LEFT_GROUP = 'left';
export const EDITOR_TOOLBAR_RIGHT_GROUP = 'right';

export const SOURCE_EDITOR_INSTANCE_ERROR_NO_EL = s__(
  'SourceEditor|"el" parameter is required for createInstance()',
);
export const ERROR_INSTANCE_REQUIRED_FOR_EXTENSION = s__(
  'SourceEditor|Source Editor instance is required to set up an extension.',
);
export const EDITOR_EXTENSION_DEFINITION_ERROR = s__(
  'SourceEditor|Extension definition should be either a class or a function',
);
export const EDITOR_EXTENSION_NO_DEFINITION_ERROR = s__(
  'SourceEditor|`definition` property is expected on the extension.',
);
export const EDITOR_EXTENSION_DEFINITION_TYPE_ERROR = s__(
  'SourceEditor|Extension definition should be either class, function, or an Array of definitions.',
);
export const EDITOR_EXTENSION_NOT_SPECIFIED_FOR_UNUSE_ERROR = s__(
  'SourceEditor|No extension for unuse has been specified.',
);
export const EDITOR_EXTENSION_NOT_REGISTERED_ERROR = s__('SourceEditor|%{name} is not registered.');
export const EDITOR_EXTENSION_NAMING_CONFLICT_ERROR = s__(
  'SourceEditor|Name conflict for "%{prop}()" method.',
);
export const EDITOR_EXTENSION_STORE_IS_MISSING_ERROR = s__(
  'SourceEditor|Extensions Store is required to check for an extension.',
);

//
// EXTENSIONS' CONSTANTS
//

// Source Editor Base Extension
export const EXTENSION_BASE_LINE_LINK_ANCHOR_CLASS = 'link-anchor';
export const EXTENSION_BASE_LINE_NUMBERS_CLASS = 'line-numbers';

// For CI config schemas the filename must match
// '*.gitlab-ci.yml' regardless of project configuration.
// https://gitlab.com/gitlab-org/gitlab/-/issues/293641
export const EXTENSION_CI_SCHEMA_FILE_NAME_MATCH = '.gitlab-ci.yml';

export const EXTENSION_MARKDOWN_PREVIEW_PANEL_CLASS = 'md';
export const EXTENSION_MARKDOWN_PREVIEW_PANEL_PARENT_CLASS = 'source-editor-preview';
export const EXTENSION_MARKDOWN_PREVIEW_ACTION_ID = 'markdown-preview';
export const EXTENSION_MARKDOWN_PREVIEW_HIDE_ACTION_ID = 'markdown-preview-hide';
export const EXTENSION_MARKDOWN_PREVIEW_PANEL_WIDTH = 0.5; // 50% of the width
export const EXTENSION_MARKDOWN_PREVIEW_UPDATE_DELAY = 250; // ms
export const EXTENSION_MARKDOWN_PREVIEW_LABEL = __('Preview Markdown');
export const EXTENSION_MARKDOWN_HIDE_PREVIEW_LABEL = __('Hide Live Preview');
export const EXTENSION_MARKDOWN_BUTTONS = [
  {
    id: 'bold',
    label: sprintf(s__('MarkdownEditor|Add bold text (%{modifierKey}B)'), {
      modifierKey,
    }),
    data: {
      mdTag: '**',
      mdShortcuts: '["mod+b"]',
    },
  },
  {
    id: 'italic',
    label: sprintf(s__('MarkdownEditor|Add italic text (%{modifierKey}I)'), {
      modifierKey,
    }),
    data: {
      mdTag: '_',
      mdShortcuts: '["mod+i"]',
    },
  },
  {
    id: 'strikethrough',
    label: sprintf(s__('MarkdownEditor|Add strikethrough text (%{modifierKey}â‡§X)'), {
      modifierKey,
    }),
    data: {
      mdTag: '~~',
      mdShortcuts: '["mod+shift+x]',
    },
  },
  {
    id: 'quote',
    label: __('Insert a quote'),
    data: {
      mdTag: '> ',
      mdPrepend: true,
    },
  },
  {
    id: 'code',
    label: __('Insert code'),
    data: {
      mdTag: '`',
      mdBlock: '```',
    },
  },
  {
    id: 'link',
    label: sprintf(s__('MarkdownEditor|Add a link (%{modifier_key}K)'), {
      modifierKey,
    }),
    data: {
      mdTag: '[{text}](url)',
      mdSelect: 'url',
      mdShortcuts: '["mod+k"]',
    },
  },
  {
    id: 'list-bulleted',
    label: __('Add a bullet list'),
    data: {
      mdTag: '- ',
      mdPrepend: true,
    },
  },
  {
    id: 'list-numbered',
    label: __('Add a numbered list'),
    data: {
      mdTag: '1. ',
      mdPrepend: true,
    },
  },
  {
    id: 'list-task',
    label: __('Add a checklist'),
    data: {
      mdTag: '- [ ] ',
      mdPrepend: true,
    },
  },
  {
    id: 'details-block',
    label: __('Add a collapsible section'),
    data: {
      mdTag: '<details><summary>Click to expand</summary>\n{text}\n</details>',
      mdPrepend: true,
      mdSelect: __('Click to expand'),
    },
  },
  {
    id: 'table',
    label: __('Add a table'),
    data: {
      /* eslint-disable-next-line @gitlab/require-i18n-strings */
      mdTag: '| header | header |\n| ------ | ------ |\n|        |        |\n|        |        |',
      mdPrepend: true,
    },
  },
];
