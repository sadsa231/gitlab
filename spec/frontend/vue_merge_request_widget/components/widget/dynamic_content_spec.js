import { shallowMountExtended } from 'helpers/vue_test_utils_helper';
import { EXTENSION_ICONS } from '~/vue_merge_request_widget/constants';
import DynamicContent from '~/vue_merge_request_widget/components/widget/dynamic_content.vue';

describe('~/vue_merge_request_widget/components/widget/dynamic_content.vue', () => {
  let wrapper;

  const createComponent = ({ propsData } = {}) => {
    wrapper = shallowMountExtended(DynamicContent, {
      propsData: {
        widgetName: 'MyWidget',
        ...propsData,
      },
      stubs: {
        DynamicContent,
      },
    });
  };

  it('renders given data', () => {
    createComponent({
      propsData: {
        data: {
          id: 'row-id',
          header: ['This is a header', 'This is a subheader'],
          text: 'Main text for the row',
          subtext: 'Optional: Smaller sub-text to be displayed below the main text',
          helpPopover: {
            options: { title: 'Widget help popover title' },
            content: { text: 'Widget help popover content' },
          },
          icon: {
            name: EXTENSION_ICONS.success,
          },
          badge: {
            text: 'Badge is optional. Text to be displayed inside badge',
            variant: 'info',
          },
          link: {
            text: 'Optional link to display after text',
            href: 'https://gitlab.com',
          },
          children: [
            {
              id: 'row-id-2',
              header: 'Child row header',
              text: 'This is recursive. It will be listed in level 3.',
            },
          ],
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
