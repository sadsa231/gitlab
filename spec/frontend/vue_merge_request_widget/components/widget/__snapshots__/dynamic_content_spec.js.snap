// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`~/vue_merge_request_widget/components/widget/dynamic_content.vue renders given data 1`] = `
"<content-row-stub level=\\"2\\" statusiconname=\\"success\\" widgetname=\\"MyWidget\\" header=\\"This is a header,This is a subheader\\" helppopover=\\"[object Object]\\" actionbuttons=\\"\\">
  <div class=\\"gl-display-flex gl-flex-direction-column\\">
    <div>
      <p class=\\"gl-mb-0\\">Main text for the row</p>
      <gl-link-stub href=\\"https://gitlab.com\\">Optional link to display after text</gl-link-stub>
      <!---->
      <gl-badge-stub size=\\"md\\" variant=\\"info\\" iconsize=\\"md\\">
        Badge is optional. Text to be displayed inside badge
      </gl-badge-stub>
      <actions-stub widget=\\"MyWidget\\" tertiarybuttons=\\"\\" class=\\"gl-ml-auto gl-pl-3\\"></actions-stub>
      <p class=\\"gl-m-0 gl-font-sm\\">Optional: Smaller sub-text to be displayed below the main text</p>
    </div>
    <ul class=\\"gl-m-0 gl-p-0 gl-list-style-none\\">
      <li>
        <content-row-stub level=\\"3\\" statusiconname=\\"\\" widgetname=\\"MyWidget\\" header=\\"Child row header\\" actionbuttons=\\"\\" data-qa-selector=\\"child_content\\">
          <div class=\\"gl-display-flex gl-flex-direction-column\\">
            <div>
              <p class=\\"gl-mb-0\\">This is recursive. It will be listed in level 3.</p>
              <!---->
              <!---->
              <!---->
              <actions-stub widget=\\"MyWidget\\" tertiarybuttons=\\"\\" class=\\"gl-ml-auto gl-pl-3\\"></actions-stub>
              <!---->
            </div>
            <!---->
          </div>
        </content-row-stub>
      </li>
    </ul>
  </div>
</content-row-stub>"
`;
